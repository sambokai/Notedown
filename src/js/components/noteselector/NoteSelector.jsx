import React from 'react';
import PropTypes from 'prop-types';
import NoteList from './notelist/NoteList';
import NotesListFilterSearch from './notesearch/NotesListFilterSearch';

class NoteSelector extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      searchQuery: '',
    };
  }

  handleSearchQueryUpdate = (searchQuery) => {
    this.setState({ searchQuery: searchQuery.trim() });
  };


  render() {
    return (
      <div>
        <NotesListFilterSearch onUpdateSearchQuery={this.handleSearchQueryUpdate} />
        <NoteList
          notes={this.props.notes}
          onSelectNote={this.props.onSelectNote}
          selectedNote={this.props.selectedNote}
          searchQuery={this.state.searchQuery}
        />
      </div>
    );
  }
}

NoteSelector.propTypes = {
  notes: PropTypes.arrayOf(PropTypes.object).isRequired,
  onSelectNote: PropTypes.func.isRequired,
  selectedNote: PropTypes.number.isRequired,
};

export default NoteSelector;
