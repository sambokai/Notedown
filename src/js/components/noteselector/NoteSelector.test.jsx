import React from 'react';

import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import NoteSelector from './NoteSelector';


const onSelectNoteMock = jest.fn();
let mockProps = {
  notes: [
    { body: 'This is Note#1', id: 1, lastChange: 1514761200000 },
    { body: 'This is Note#2', id: 2, lastChange: 1514847600000 },
    { body: 'This is Note#3', id: 3, lastChange: 1514934000000 },
    { body: 'This is Note#4', id: 4, lastChange: 1515020400000 },
  ],
  onSelectNote: onSelectNoteMock,
};
mockProps = { ...mockProps, selectedNote: mockProps.notes[2] };

describe('<NoteSelector/>', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<NoteSelector {...mockProps} />);

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('handleSearchQueryUpdate(searchQuery)', () => {
  it('should trim trailing and leading whitespace/newlines of searchQuery state', () => {
    const wrapper = shallow(<NoteSelector {...mockProps} />);
    const searchQuery = '  Bananas  \n';

    wrapper.instance().handleSearchQueryUpdate(searchQuery);

    expect(wrapper.state('searchQuery')).toBe('Bananas');
  });
});
