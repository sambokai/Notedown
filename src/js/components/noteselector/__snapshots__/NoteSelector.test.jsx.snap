// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoteSelector/> should render correctly 1`] = `
<div
  className="mh-100 col-3 pr-2"
>
  <NotesListFilterSearch
    onUpdateSearchQuery={[Function]}
  />
  <NoteList
    notes={
      Array [
        Object {
          "body": "This is Note#1",
          "id": 1,
          "lastChange": 1514761200000,
        },
        Object {
          "body": "This is Note#2",
          "id": 2,
          "lastChange": 1514847600000,
        },
        Object {
          "body": "This is Note#3",
          "id": 3,
          "lastChange": 1514934000000,
        },
        Object {
          "body": "This is Note#4",
          "id": 4,
          "lastChange": 1515020400000,
        },
      ]
    }
    searchQuery=""
  />
</div>
`;
